### ISSUE TITLE
bluegreen_deploy Client Error: Conflict for url

### ISSUE THREAD

**Issue by cgmartin** [mesosphere/marathon-lb#190]
Hello, I'm running a single VM with Mesos/Marathon/Marathon-lb and getting the following error during a bluegreen_deploy. Any ideas?

```
$ docker run -it -v ${PWD}:/tmp/deploy --entrypoint /bin/bash \
mesosphere/marathon-lb -c "./bluegreen_deploy.py --json /tmp/deploy/microbot-example.json \
--marathon http://10.0.0.3:8080 --marathon-lb http://10.0.0.3:9090 \
--syslog-socket /dev/null --force"

bluegreen_deploy: Final app definition:
{
  "container": {
    "docker": {
      "image": "dontrebootme/microbot:v2",
      "network": "BRIDGE",
      "portMappings": [
        {
          "containerPort": 80,
          "hostPort": 0,
          "protocol": "tcp",
          "servicePort": 8001
        }
      ]
    },
    "type": "DOCKER"
  },
  "cpus": 0.03,
  "healthChecks": [
    {
      "gracePeriodSeconds": 10,
      "intervalSeconds": 30,
      "path": "/",
      "protocol": "HTTP"
    }
  ],
  "id": "/microbot-green",
  "instances": 1,
  "labels": {
    "HAPROXY_0_PORT": "8000",
    "HAPROXY_APP_ID": "microbot",
    "HAPROXY_DEPLOYMENT_ALT_PORT": "8001",
    "HAPROXY_DEPLOYMENT_COLOUR": "green",
    "HAPROXY_DEPLOYMENT_GROUP": "microbot",
    "HAPROXY_DEPLOYMENT_STARTED_AT": "2016-05-13T16:37:31.438223",
    "HAPROXY_DEPLOYMENT_TARGET_INSTANCES": "2",
    "HAPROXY_GROUP": "external"
  },
  "mem": 64.0,
  "upgradeStrategy": {
    "maximumOverCapacity": 0.1,
    "minimumHealthCapacity": 0.8
  }
}
bluegreen_deploy: Existing app running 2 instances, new app running 1 instances
bluegreen_deploy: Found 2 app listeners across 1 HAProxy instances
bluegreen_deploy: Existing app running 2 instances, new app running 1 instances
bluegreen_deploy: Found 2 app listeners across 1 HAProxy instances
bluegreen_deploy: Existing app running 2 instances, new app running 1 instances
bluegreen_deploy: Found 2 app listeners across 1 HAProxy instances
bluegreen_deploy: Existing app running 2 instances, new app running 1 instances
bluegreen_deploy: Found 2 app listeners across 1 HAProxy instances
bluegreen_deploy: Existing app running 2 instances, new app running 1 instances
bluegreen_deploy: Found 2 app listeners across 1 HAProxy instances
bluegreen_deploy: Existing app running 2 instances, new app running 1 instances
bluegreen_deploy: Found 3 app listeners across 1 HAProxy instances
bluegreen_deploy: Existing app running 2 instances, new app running 1 instances
bluegreen_deploy: Found 3 app listeners across 1 HAProxy instances
bluegreen_deploy: Existing app running 2 instances, new app running 1 instances
bluegreen_deploy: Found 3 app listeners across 1 HAProxy instances
bluegreen_deploy: Existing app running 2 instances, new app running 1 instances
bluegreen_deploy: Found 3 app listeners across 1 HAProxy instances
bluegreen_deploy: Existing app running 2 instances, new app running 1 instances
bluegreen_deploy: Found 3 app listeners across 1 HAProxy instances
bluegreen_deploy: There are 0 drained listeners, about to kill & scale for these tasks:
  - 
bluegreen_deploy: Scaling new app up to 2 instances
bluegreen_deploy: Scaling down old app by 0 instances
bluegreen_deploy: Existing app running 2 instances, new app running 2 instances
bluegreen_deploy: Found 4 app listeners across 1 HAProxy instances
bluegreen_deploy: There are 0 drained listeners, about to kill & scale for these tasks:
  - 
bluegreen_deploy: Scaling new app up to 2 instances
Traceback (most recent call last):
  File "./bluegreen_deploy.py", line 613, in <module>
    if do_bluegreen_deploy(args):
  File "./bluegreen_deploy.py", line 528, in do_bluegreen_deploy
    return swap_bluegreen_apps(args, new_app, old_app, time.time())
  File "./bluegreen_deploy.py", line 282, in swap_bluegreen_apps
    scale_new_app_instances(args, new_app, old_app)
  File "./bluegreen_deploy.py", line 317, in scale_new_app_instances
    return scale_marathon_app_instances(args, new_app, instances)
  File "./bluegreen_deploy.py", line 355, in scale_marathon_app_instances
    response.raise_for_status()
  File "/usr/local/lib/python3.5/dist-packages/requests/models.py", line 844, in raise_for_status
    raise HTTPError(http_error_msg, response=self)
requests.exceptions.HTTPError: 409 Client Error: Conflict for url: http://10.0.0.3:8080/v2/apps/microbot-green
```

#### Versions
Mesos: 0.26.0
Marathon: 0.15.0
Marathon-lb: latest / f24919997631 (Image ID)

**Comment by brndnmtthws** [comment]
Strange. I wonder if there was a pending deployment for some reason. Normally, however, it should wait until all new tasks are alive and well.

**Comment by brndnmtthws** [comment]
I'd suggest, by the way, that you run the script again with `-r`, which will resume where it left off.

**Comment by JamesAwesome** [comment]
@brndnmtthws it looks like this happens when you try to scale the app while a marathon deployment is already in progress... (eg: app was in the midst of scaling from 2 to 4, you told it to go from 4 to 6, marathon says 409 bother me after i finish going from 2 to 4)

**Comment by brndnmtthws** [comment]
This should be resolved by #192.
